### DISCUSSION TITLE
Provide the means to globally override the default 422 OpenAPI schema

### DISCUSSION THREAD

**Discussion by skeletorXVI** [fastapi/fastapi#8775] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Body
from fastapi.exceptions import RequestValidationError
from pydantic import BaseModel
from starlette import status
from starlette.requests import Request
from starlette.responses import JSONResponse


class Example(BaseModel):
    value: int


class Error(BaseModel):
    message: str


app = FastAPI()


@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
    error = Error(message="Validation failed")
    return JSONResponse(error.json(), status_code=status.HTTP_422_UNPROCESSABLE_ENTITY)


@app.post("/", response_model=Example)
def root(example: Example = Body(...)):
    return example
```

### Description

Overriding the default exception handlers is easy and [well documented](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-the-httpexception-error-handler). Unfortunately this has a significant caveat in regards to the generated OpenAPI specifications, the schema does not match the actual responses.

### Wanted Solution

Provide a parameter or function that can override the default 422 schema model.

### Wanted Code

```python
from fastapi import FastAPI, Body
from fastapi.exceptions import RequestValidationError
from pydantic import BaseModel
from starlette import status
from starlette.requests import Request
from starlette.responses import JSONResponse


class Example(BaseModel):
    value: int


class Error(BaseModel):
    message: str


# Provide error model as parameter
app = FastAPI(
    validation_error_model=Error,
    validation_error_description="Returns an object with a simple error message",
    validation_error_status=400,
)


# or set via fields
app.validation_error_model = Error
app.validation_error_description = "Returns an object with a simple error message"
app.validation_error_status = 400


@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
    error = Error(message="Validation failed")
    return JSONResponse(error.json(), status_code=status.HTTP_422_UNPROCESSABLE_ENTITY)


@app.post("/", response_model=Example, responses={422: {"model": Error}})
def root(example: Example = Body(...)):
    return example
```

### Alternatives

Manually add a response model for 422 to every endpoint that can raise a RequestValidationError. This is suitable for a small API but will lead to problems as the application grows and maintenance is required.

### Operating System

Linux

### Operating System Details

DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=21.0.7
DISTRIB_CODENAME=Ornara
DISTRIB_DESCRIPTION="Manjaro Linux"

### FastAPI Version

0.67.0

### Python Version

Python 3.9.5

**Comment by Kludex** [comment]
I agree we should have something easier, but for the time being: https://github.com/tiangolo/fastapi/issues/2455#issuecomment-736876736

**Comment by caiofauza** [comment]
Nice enhancement. We will try to work on it.
@Gustavobb
@zMendes

**Comment by mvanderlee** [comment, marked as answer]
@skeletorXVI This can now be accomplished by defining your model using the `responses` field. I had the same problem and found this discussion.

```python
app = FastAPI(
    responses={422: {"model": YourErrorModel}}
)
```
